### ISSUE TITLE
Should we skip the User-Agent header?

### ISSUE THREAD

**Issue by FantasticFiasco** [FantasticFiasco/aws-signature-version-4#1208]
**Describe the bug**

As described [a review remark of #1207](https://github.com/FantasticFiasco/aws-signature-version-4/pull/1207#discussion_r1864568014), let's analyze if we should skip the `User-Agent` header instead having special support for it in the codebase (#1155).

**Comment by mungojam** [comment]
I wouldn't call it a workaround, it was changing it to implement [the user agent spec](https://en.wikipedia.org/wiki/User-Agent_header#Format_for_human-operated_web_browsers). But if it can be skipped altogether in the signature then that's a good result

**Comment by FantasticFiasco** [comment]
You are correct, my intention was not to downplay this contribution. Bad wording on my part, I'll update my text.

**Comment by FantasticFiasco** [comment]
As it currently is worded, do you think it makes more sense, or can it be improved?

**Comment by mungojam** [comment]
> As it currently is worded, do you think it makes more sense, or can it be improved?

It's all good now thanks 🙂

**Comment by cfbao** [comment]
I found one case where it makes sense to skip the user-agent header:
If a user of this library adds an empty string as the user-agent header (not implausible in a proxy server implementation that calls [`TryAddWithoutValidation`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.tryaddwithoutvalidation) on almost all headers), and sends the signed request to API Gateway, AWS will change the user-agent header to "Amazon CloudFront", and then the signature will be invalid.
